src/test/regress/sql/schema_based_sharding.sql

```diff
@@ -0,0 +1,700 @@
+EXECUTE list_tuples;
+
+SET search_path TO tenant_7;
+DROP TABLE search_path_test;
+SELECT * FROM pg_dist_partition WHERE logicalrelid::text = 'search_path_test';
+
+SET search_path TO tenant_5;
+SELECT regular_schema.decrement_one();
+EXECUTE list_tuples;
+
+SET search_path TO regular_schema;
+
+CREATE USER test_other_super_user WITH superuser;
+
+\c - test_other_super_user
+
+SET citus.enable_schema_based_sharding TO ON;
+CREATE SCHEMA tenant_9;
+
+\c - postgres
+
+SET search_path TO regular_schema;
+SET citus.next_shard_id TO 1930000;
+SET citus.shard_count TO 32;
+SET citus.shard_replication_factor TO 1;
+SET client_min_messages TO NOTICE;
+SET citus.enable_schema_based_sharding TO ON;
+
+SELECT schemaid AS tenant_9_schemaid FROM pg_dist_tenant_schema WHERE schemaid::regnamespace::text = 'tenant_9' \gset
+SELECT colocationid AS tenant_9_colocationid FROM pg_dist_tenant_schema WHERE schemaid::regnamespace::text = 'tenant_9' \gset
+
+SELECT result FROM run_command_on_workers($$
+    SELECT schemaid INTO tenant_9_schemaid FROM pg_dist_tenant_schema
+    WHERE schemaid::regnamespace::text = 'tenant_9'
+$$);
+
+DROP OWNED BY test_other_super_user;
+
+-- (on coordinator) Make sure that dropping an empty tenant schema
+-- (via DROP OWNED BY) doesn't leave any dangling entries in
+-- pg_dist_tenant_schema and pg_dist_colocation.
+SELECT COUNT(*)=0 FROM pg_dist_tenant_schema WHERE schemaid = :tenant_9_schemaid;
+SELECT COUNT(*)=0 FROM pg_dist_colocation WHERE colocationid = :tenant_9_colocationid;
+
+-- (on workers) Make sure that dropping an empty tenant schema
+-- (via DROP OWNED BY) doesn't leave any dangling entries in
+-- pg_dist_tenant_schema and pg_dist_colocation.
+SELECT result FROM run_command_on_workers($$
+    SELECT COUNT(*)=0 FROM pg_dist_tenant_schema
+    WHERE schemaid = (SELECT schemaid FROM tenant_9_schemaid)
+$$);
+
+SELECT format(
+    'SELECT result FROM run_command_on_workers($$
+        SELECT COUNT(*)=0 FROM pg_dist_colocation WHERE colocationid = %s;
+    $$);',
+:tenant_9_colocationid) AS verify_workers_query \gset
+
+:verify_workers_query
+
+SELECT result FROM run_command_on_workers($$
+    DROP TABLE tenant_9_schemaid
+$$);
+
+DROP USER test_other_super_user;
+
+CREATE ROLE test_non_super_user WITH LOGIN;
+ALTER ROLE test_non_super_user NOSUPERUSER;
+
+GRANT CREATE ON DATABASE regression TO test_non_super_user;
+SELECT result FROM run_command_on_workers($$GRANT CREATE ON DATABASE regression TO test_non_super_user$$);
+
+GRANT CREATE ON SCHEMA public TO test_non_super_user ;
+
+\c - test_non_super_user
+
+SET search_path TO regular_schema;
+SET citus.next_shard_id TO 1940000;
+SET citus.shard_count TO 32;
+SET citus.shard_replication_factor TO 1;
+SET client_min_messages TO NOTICE;
+SET citus.enable_schema_based_sharding TO ON;
+
+-- test create / drop tenant schema / table
+
+CREATE SCHEMA tenant_10;
+CREATE TABLE tenant_10.tbl_1(a int, b text);
+CREATE TABLE tenant_10.tbl_2(a int, b text);
+
```

Comment:
consider adding a `CREATE TABLE tenant_10.tbl_3 AS SELECT s, s FROM generate_series(1,100) s;` test

This currently results in a local table being created in the schema.

Reply:
now we're throwing an error for that